Source/Initialization/WarpXInitData.cpp

```diff
@@ -211,6 +213,65 @@ WarpX::ComputePMLFactors ()
     }
 }
 
+void
+WarpX::ComputeMaxStep ()
+{
+    if (do_compute_max_step_from_zmax) {
+        computeMaxStepBoostAccelerator(geom[0]);
+    }
+
+    max_step = std::min(max_step,
+                        istep[0] + static_cast<int>(std::ceil( (stop_time-t_new[0])/dt[0] )));
```

Comment:
This should make it so that `max_step` is consistent with `stop_time`, which is convenient if we need to get the last timestep of the simulation from another part of the code.

Is there any reason not to do that? (maybe we want to use e.g. non constant timesteps in the future?)